plugin-code/src/main/java/com/trivago/gherkin/GherkinDocumentParser.java

```diff
@@ -353,7 +346,35 @@ private final boolean scenarioShouldBeIncluded(final List<String>... sourceTagsL
         } catch (TagExpressionException e) {
             throw new CucablePluginException("The tag expression '" + includeScenarioTags + "' is invalid: " + e.getMessage());
         }
-        return tagExpression.evaluate(combinedScenarioTags);
+        return tagExpression.evaluate(combinedScenarioTags) && scenarioNameMatchExists;
+    }
+
+    /**
+     * Checks if a scenarioName value matches with the scenario name.
+     *
+     * @param stringToMatch the string that will be matched with the scenarioName value.
+     * @return index of the scenarioName value in the scenarioNames list if a match exists.
+     *         -1 if no match exists.
+     */
+    public int matchScenarioWithScenarioNames(String stringToMatch) {
+        List<String> scenarioNames = propertyManager.getScenarioNames();
+        int matchIndex = -1;
+
+        if (scenarioNames == null || scenarioNames.isEmpty()) {
+            return 0;
+        }
+
+        for (String scenarioName : scenarioNames) {
+            String regex = "(Scenario|Example):.+" + scenarioName;
```

Comment:
This will not work with non-english scenarios/examples.

Reply:
Completely forgot about the non-english part. Sorry! I will look into this.

Reply:
I have checked in the changes. Please take a look.